nexus/src/app/sagas/instance_create.rs

```diff
@@ -428,38 +428,42 @@ async fn sic_create_network_interface_undo(
         .lookup_for(authz::Action::Modify)
         .await
         .map_err(ActionError::action_failed)?;
-    match LookupPath::new(&opctx, &datastore)
+
+    let interface_deleted = match LookupPath::new(&opctx, &datastore)
         .instance_network_interface_id(interface_id)
         .lookup_for(authz::Action::Delete)
         .await
     {
         Ok((.., authz_interface)) => {
+            // The lookup succeeded, but we could still fail to delete the
+            // interface if we're racing another deleter.
             datastore
                 .instance_delete_network_interface(
                     &opctx,
                     &authz_instance,
                     &authz_interface,
                 )
                 .await
-                .map_err(|e| e.into_external())?;
-            Ok(())
+                .map_err(|e| e.into_external())?
```

Comment:
Hmm, so we could still return `ObjectNotFound` here (NIC doesn't exist at all) thus skipping the warning below. In theory we shouldn't end up there since the authz lookup should fail first and we don't do hard deletes. I guess failing the saga makes sense at that point.

Reply:
I think that's right, yeah - an `ObjectNotFound` on this call means the NIC ID has either been hard deleted or never existed, neither of which seems like a "just warn and keep going" case.